--- request.php	2014-11-10 16:18:34.000000000 +0000
+++ request.new	2015-01-16 00:33:39.079018406 +0000
@@ -43,7 +43,38 @@
 		if (preg_match(self::REGEX_LOCALHOST, $domain) === 1) {
 			return true;
 		}
-		return in_array($domain, $trustedList);
+
+		// Allow access from an explicitly listed domain
+		if( in_array($domain, $trustedList)) {
+			return true;
+		}
+
+		// If a value contains a *, apply glob-style matching. Any second * is ignored.
+		foreach ($trustedList as $trusted) {
+			if($trusted == '*') {
+				return true;
+			}
+			$star = strpos($trusted, '*');
+			if($star === false) {
+				next;
+			}
+			if($star === 0) {
+				if(strrpos($domain, substr($trusted, 1)) !== FALSE) {
+					return true;
+				}
+                        } elseif($star === strlen($trusted)-1) {
+				if(strpos($domain, substr($trusted, 0, strlen($trusted)-1 )) !== FALSE) {
+					return true;
+				}
+			} else {
+				if(strpos($domain, substr($trusted, 0, $star)) !== false
+				&& strrpos($domain, substr($trusted, $star+1 ), -strlen($trusted-$star-1)) !== false )
+				{
+					return true;
+				}
+			}
+		}
+		return false;
 	}
 
 	/**
